### PR TITLE
Fix update props and connectWpPost slug entity resolution

### PR DESCRIPTION
**Pull request by sdgluck**
Fixes bugs identified in #44 by @olapersson that prevented component updating on changed props and connectWpPost components resolving entities by slug.

Extras: locks down deps, `shouldUpdate` can take `state` as third arg, small updates to `README`, and others I have forgotten. 💯

### CODE REVIEW

File: src/connect.js [hunk 1]
```diff
@@ -191,32 +194,36 @@ export function connectWpPost (contentType, id) {
...
+
         if (!query || !query.complete || query.error) return null
+
+        const entities = this.props.wordpress.entities[typeConfig.plural]
+        const keys = Object.keys(entities)
         const realId = identifier(displayName, id, props)
-        const entities = this._getState().wordpress.entities[typeConfig.plural]
-        return entities[realId] || null
+
+        for (let i = 0, len = keys.length; i < len; i++) {
```
Comment (rasmuswinter) [inline src/connect.js]: As discussed earlier, can't we use the result of the query to know which ID(s) were returned, and then look them up in the entities array?

Comment (sdgluck) [inline src/connect.js]: Some content types, e.g. tag, don't have an `id` so for these types we fall back on their `slug`, and vice versa. Maybe there's a better way still?

Comment (rasmuswinter) [inline src/connect.js]: But this is part of `KasiaConnectWpPostComponent`, so won't the data always be a post, so have an id?

Comment (sdgluck) [inline src/connect.js]: Ah, no, sorry... confusingly post is synonymous with content type here.

Comment (rasmuswinter) [inline src/connect.js]: Ah, I didn't realise that. In that case, this should do until (unless) we find a better solution

**Pull request merged**